cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/BufferingTable.java

```diff
@@ -406,6 +426,52 @@ public Row get(byte[] row, byte[] startColumn, byte[] stopColumn, int limit) {
     }
   }
 
+  @Override
+  public List<Row> get(List<Get> gets) {
+    try {
+      // get persisted, then overwrite with whats buffered
+      List<Map<byte[], byte[]>> persistedRows = getPersisted(gets);
+      // gets and rows lists are always of the same size
+      Preconditions.checkArgument(gets.size() == persistedRows.size(),
+        "Invalid number of rows fetched when performing multi-get. There must be one row for each get.");
+
+      List<Row> result = Lists.newArrayListWithCapacity(persistedRows.size());
+
+      Iterator<Map<byte[], byte[]>> persistedRowsIter = persistedRows.iterator();
+      Iterator<Get> getIter = gets.iterator();
+      while (persistedRowsIter.hasNext() && getIter.hasNext()) {
+        Get get = getIter.next();
+        Map<byte[], byte[]> persistedRow = persistedRowsIter.next();
+
+        // navigable copy of the persisted data. Implementation may return immutable or unmodifiable maps,
+        // so we make a copy here.
+        NavigableMap<byte[], byte[]> rowColumns = Maps.newTreeMap(Bytes.BYTES_COMPARATOR);
+        rowColumns.putAll(persistedRow);
+
+        byte[] row = get.getRow();
+        NavigableMap<byte[], Update> buffCols = buff.get(row);
+        // if the row was deleted in the buffer, add an empty row to the results
+        if (buffCols == null && buff.containsKey(row)) {
```

Comment:
I see a similar check in the get(Get) method, but I don't think this is actually needed, as we never seem to put a null value for the entire row.  Instead the delete(byte[] row) method first retrieves all columns, then puts a null value for each column.